python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1365,40 +1378,56 @@ def can_run_with_parent(
             and partition_mapping.end_offset == 0
         )
     )
-    return (
-        parent_node.backfill_policy == candidate_node.backfill_policy
-        and parent_node.priority_repository_handle is candidate_node.priority_repository_handle
-        and parent_node.partitions_def == candidate_node.partitions_def
-        and (
-            parent.partition_key in asset_partitions_to_request_map[parent.asset_key]
-            or parent in candidates_unit
+    if parent_node.backfill_policy != candidate_node.backfill_policy:
+        not_requested_reasons[candidate] = (
+            f"parent {parent_node.key} and {candidate_node.key} have different backfill policies"
         )
-        and (
-            # if there is a simple mapping between the parent and the child, then
-            # with the parent
-            has_identity_partition_mapping
-            # if there is not a simple mapping, we can only materialize this asset with its
-            # parent if...
-            or (
-                # there is a backfill policy for the parent
-                parent_node.backfill_policy is not None
-                # the same subset of parents is targeted as the child
-                and parent_target_subset.value == candidate_target_subset.value
-                and (
-                    # there is no limit on the size of a single run or...
-                    parent_node.backfill_policy.max_partitions_per_run is None
-                    # a single run can materialize all requested parent partitions
-                    or parent_node.backfill_policy.max_partitions_per_run
-                    > len(asset_partitions_to_request_map[parent.asset_key])
-                )
-                # all targeted parents are being requested this tick
-                and len(asset_partitions_to_request_map[parent.asset_key])
-                == parent_target_subset.size
+        return False
+    if parent_node.priority_repository_handle is not candidate_node.priority_repository_handle:
+        not_requested_reasons[candidate] = (
+            f"parent {parent_node.key} and {candidate_node.key} are in different code locations"
+        )
+        return False
```

Comment:
might be cleaner to return the reason instead of mutating a passed in structure since we appear to always return `True` or `False, reason`

Reply:
yeah i was thinking about that too, but the bfs function where this is called seems like it's meant to be general use, so i didn't want to put in backfill-specific stuff. i sorta went back into prototyping mode w this PR so i can mark as draft until i find a version i like